### PR TITLE
Rewrite server to just gRPC and improve perf

### PR DESCRIPTION
**Pull request by edaniels**
Still have some work left to do on packaging and testing but the important code is all done and tested from python gRPC scripts. Performance has been improved greatly (reference the slack convo). I also have not hit any crashes yet.

- Only gRPC is available; GetPointCloud still unimplemented like it was
- Improved logging and debugging
- No OpenCV
- Using libjpeg-turbo for JPEG
- Using fpng for PNG (faster than libpng but not fast)

New command line args:
`usage: intelrealgrpcserver [port_number] [color_width] [color_height]  [depth_width] [depth_height] [--disable-depth] [--disable-color]`

Sample output:

No args:
```
Intel RealSense gRPC server
note: will pick any suitable color_width and color_height
note: will pick any suitable depth_width and depth_height
found device:
name:      Intel RealSense D435
serial:    022422071219
firmware:  05.12.07.100
port:      2-1-5
usb type:  3.2
pipeline started with:
color_enabled:  true
color_width:    640
color_height:   480
depth_enabled:  true
depth_width:    640 (aligned to color)
depth_height:   480 (aligned to color)
waiting for camera frame loop thread to be ready... ready!
listening on 0.0.0.0:8085
```

`8086 0 0  0 0 --disable-color`:
```
Intel RealSense gRPC server
note: will pick any suitable color_width and color_height
note: will pick any suitable depth_width and depth_height
found device:
name:      Intel RealSense D435
serial:    022422071219
firmware:  05.12.07.100
port:      2-1-5
usb type:  3.2
pipeline started with:
color_enabled:  false
depth_enabled:  true
depth_width:    848
depth_height:   480
waiting for camera frame loop thread to be ready... ready!
listening on 0.0.0.0:8086
```

`8086 1920 1080 848 480 --debug`:
```
Intel RealSense gRPC server
found device:
name:      Intel RealSense D435
serial:    022422071219
firmware:  05.12.07.100
port:      2-1-5
usb type:  3.2
pipeline started with:
color_enabled:  true
color_width:    1920
color_height:   1080
depth_enabled:  true
depth_width:    1920 (aligned to color)
depth_height:   1080 (aligned to color)
waiting for camera frame loop thread to be ready... ready!
[frameLoop] wait for frames: 0ms
listening on 0.0.0.0:8086
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 22ms
[frameLoop] total:           40ms
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 20ms
[frameLoop] total:           39ms
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 20ms
[frameLoop] total:           40ms
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 21ms
[GetImage]  JPEG encode:     26ms
[GetImage]  total:           29ms
[frameLoop] total:           40ms
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 22ms
[GetImage]  JPEG encode:     29ms
[GetImage]  total:           29ms
[frameLoop] total:           43ms
[frameLoop] wait for frames: 0ms
[frameLoop] frame alignment: 22ms
```

cc: @tessavitabile

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,6 @@ For MacOS (M1 macs), you can use our homebrew formula from the Viam tap:
 ```
 brew install viamrobotics/brews/intel-real-grpc-server
```
Comment (edaniels) [inline README.md:16]: still working on readme updates. need to make mac work again

File: intel_realsense_grpc.cpp [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    if (err) {
+        return 1;
+    }
+    tie(depthHeight, err) = parseIntArg(5, "depth_height");
+    if (err) {
+        return 1;
+    }
+
+    if (colorWidth == 0 || colorHeight == 0) {
+        cout << "note: will pick any suitable color_width and color_height\n";
```
Comment (erh) [inline intel_realsense_grpc.cpp:460]: should use `endl` rather than "\n", here and many places, otherwise might sit in a buffer and not print

**Review by erh — APPROVED** [review]

File: intel_realsense_grpc.cpp (hunk 2 above)
Comment (edaniels) [reply intel_realsense_grpc.cpp:460]: In startup code I think endl is okay which I'll change. In other code where these are here and we're debugging, constant flushing is added pressure and in most cases it does print. I'll change the cerrs though too.

**Comment by bhaney** [comment]
One Question: We are fine with completely eliminating the HTTP source and instructions for downloading from the repo?

**Comment by edaniels** [comment]
Yep we are only wanting to maintain gRPC at this point

**Review by bhaney — APPROVED** [review]
Other than the question, looks great! Thank you

**Pull request merged**